server/src/main/java/org/elasticsearch/health/components/controller/ClusterCoordination.java

```diff
@@ -10,12 +10,15 @@
 
 import org.elasticsearch.cluster.ClusterState;
 import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodeRole;
 import org.elasticsearch.cluster.node.DiscoveryNodes;
 import org.elasticsearch.health.HealthComponentResult;
 import org.elasticsearch.health.HealthIndicatorResult;
 import org.elasticsearch.health.HealthStatus;
 
-import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public final class ClusterCoordination {
```

Comment:
I believe this should implement `HealthIndicatorService` to plug into functionality (I hope to merge https://github.com/elastic/elasticsearch/pull/83205 soon)

Reply:
Or may be we should treat indicators coming from `server` differently? cc @henningandersen